docs/tools/firmware.sh

```diff
@@ -51,46 +49,37 @@ case "$os" in
 		echo
 		echo -e "Run the following commands or run this script itself in Linux now to set up Wi-Fi :-\n\nsudo mkdir -p /tmp/apple-wifi-efi\nsudo mount /dev/nvme0n1p1 /tmp/apple-wifi-efi\nbash /tmp/apple-wifi-efi/firmware.sh\nsudo umount /tmp/apple-wifi-efi\n"
 		;;
+
 	(Linux)
 		echo "Detected Linux"
 		echo "Re-mounting the EFI partition"
 		mountpoint=$(mktemp -d)
 		workdir=$(mktemp -d)
 		echo "Installing Wi-Fi and Bluetooth firmware"
-		if [[ ${1-default} = -v ]]
-		then
-			sudo mount -v /dev/nvme0n1p1 $mountpoint
-			sudo tar --warning=no-unknown-keyword -xvC $workdir -f $mountpoint/firmware.tar.gz
-			sudo python3 $0 $workdir $workdir/firmware-renamed.tar
-			sudo tar -xvC /lib/firmware -f $workdir/firmware-renamed.tar
-		else
-			sudo mount /dev/nvme0n1p1 $mountpoint
-			sudo tar --warning=no-unknown-keyword -xC $workdir -f $mountpoint/firmware.tar.gz
-			sudo python3 $0 $workdir $workdir/firmware-renamed.tar &> /dev/null
-			sudo tar -xC /lib/firmware -f $workdir/firmware-renamed.tar
-		fi
+		sudo mount ${verbose} /dev/nvme0n1p1 $mountpoint
+		sudo tar --warning=no-unknown-keyword ${verbose} -xC $workdir -f $mountpoint/firmware.tar.gz
+		sudo python3 $0 $workdir $workdir/firmware-renamed.tar ${verbose}
+
+		# ensure /lib/firmware exists before doing this
+		sudo mkdir ${verbose} -p /lib/firmware
```

Comment:
We shouldn't do this since `/lib` (and `/usr/lib`) is usually managed by the distro, you should just make the script exit if `/lib/firmware` doesnt exist.

Reply:
That does make sense but breaks setups on NixOS. Perhaps falling back to copying the firmware to some other directory then existing would be a better solution?

Reply:
> That does make sense but breaks setups on NixOS.

NixOS can ship a custom script, its weird enough that our script will probably break something anyways. You could make it print a warning to see the wiki or something.

> Perhaps falling back to copying the firmware to some other directory then exiting would be a better solution?

This wont help much since the firmware archive is in the EFI partition, so if somebody wanted the firmware they can just extract it from `/boot/efi`.

Reply:
> NixOS can ship a custom script, its weird enough that our script will probably break something anyways. You could make it print a warning to see the wiki or something.

Alright, will see what can be done.

> This wont help much since the firmware archive is in the EFI partition, so if somebody wanted the firmware they can just extract it from `/boot/efi`.

AFAIK the original firmware stuff is unrenamed and (probably? - will test now) doesn't work OOTB. That was one of my motivations of making this PR.

Reply:
> AFAIK the original firmware stuff is unrenamed and (probably? - will test now) doesn't work OOTB. That was one of my motivations of making this PR.

The firmware archive needs renaming, but after renaming it should *just work*. The NixOS script can just be a fork of this script with NixOS specific stuff added, bt NixOS is currently unmaintained so I dont know if this will happen soon.